### ISSUE TITLE
Naruto Shippuden Ultimate Ninja Heroes 3 : Japanese audio on cutscenes - Since 1.8.0

### ISSUE THREAD

**Issue by EmiNVSGMZ** [hrydgard/ppsspp#15606]
### Game or games this happens in

ULUS-10518 - Ultimate Ninja Heroes 3

### What area of the game / PPSSPP

At the beginning and end of a chapter there's a cutscene(?, for some reason even when the language is in english the audio of those is japanese.

https://streamable.com/zg9p3m

### What should happen

Audio of cutscenes should be the same than the one in the rest of the game.

### Platform

Android

### Mobile phone model or graphics card

AArch64 processor rev 13

### PPSSPP version affected

v.1.12.3

### Last working version

v.1.7.4

### Graphics backend (3D API)

OpenGL / GLES

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [ ] Try without any cheats and without loading any save states.
- [ ] Include logs or screenshots of issue.

**Comment by unknownbrackets** [comment]
Hm, it could be something to do with video files that have two audio streams and something wrong about the selection of the stream.

It would help if you could try multiple versions.  Those are pretty old though.  If you have access to a PC, it would help to start with by figuring out the first stable version affected, and the last working version (uoi say v1.7.4, but does that mean v1.7.5 or v1.8.0 is affected?)

Is there any demo of this game that has the same issue?

-[Unknown]

**Comment by Panderner** [comment]
The Last Build I tested is 1.3 as other versions won't work until 1.8.0 as I Mentioned in #9591. 1.3 had English audio working in intro cutscene, but 1.8.0 had to set japanese audio in intro cutscene even if the voice setting is set to English.

**Comment by EmiNVSGMZ** [comment]
> It would help if you could try multiple versions. Those are pretty old though. If you have access to a PC, it would help to start with by figuring out the first stable version affected, and the last working version (uoi say v1.7.4, but does that mean v1.7.5 or v1.8.0 is affected?)

I'm talking about the 1.7.4 provided here, anything above that has the audio wrong, anything below is fine (Besides the black screen, which i used a cheat to bypass).

https://www.ppsspp.org/downloads_all.html

**Comment by EmiNVSGMZ** [comment]
> The Last Build I tested is 1.3 as other versions won't work until 1.8.0 as I Mentioned in #9591.

You can use the cheat provided there to skip the black screen, versions above 1.3 have no audio problem up to 1.7.4, 1.8.0 is where it starts.

**Comment by Panderner** [comment]
Yeah I know that cheat skips the Criware and CyberConnect2 logo that caused black screen for between 1.4 to 1.7.4, 1.7.4 would be the last version work.

**Comment by Sukotto-1999** [comment]
To clarify, I was able to try investigating the thing that's causing the issue. This game uses multitrack audio for the game's cutscenes (excluding the Japanese version) to show proof, I've demultiplexed one of the PMF files from the game and i can confirm the audio in the .pmf files do have multitrack audio, one in Japanese, and the other in English.

**Comment by hrydgard** [comment]
Okay, I guess we don't support multitrack properly then and just end up picking the wrong track...

**Comment by unknownbrackets** [comment]
Well, it's probably related to #6384.  I think I had found a multitrack audio to test against, but it could be it's broken since then.  That was in v0.9.9.

-[Unknown]

**Comment by sum2012** [comment]
PPSSPP's MpegDemux uses a single shared queue for all audio channels. When the game called sceMpegGetAtracAu for the Japanese stream (even if it intended to skip it), it triggered a demux operation that pushed Japanese audio data into the shared queue. Subsequent calls for the English stream pushed English data after the Japanese data. The final Decode call simply popped the next frame from the queue, which was the Japanese one. Gemini said that even fix this game , still have a bug. I need to do more thing.
